linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/NameGen.scala

```diff
@@ -107,7 +110,13 @@ private[emitter] final class NameGen {
   }
 
   def genName(name: LabelName): String = genNameGeneric(name, genLabelNameCache)
-  def genName(name: FieldName): String = genNameGeneric(name, genFieldNameCache)
+  def genName(name: SimpleFieldName): String = genNameGeneric(name, genSimpleFieldNameCache)
+
+  def genName(name: FieldName): String = {
+    genFieldNameCache.getOrElseUpdate(name, {
+      genName(name.className) + "__f_" + genName(name.simpleName)
+    })
+  }
```

Comment:
I would not cache this for now, to avoid changing performance behavior. (Also see comments RE duplicate caching of simple field names).